lib/tools/system-calls.js

```diff
@@ -793,6 +793,7 @@ systemCallMethods.reboot = async function (retries = DEFAULT_ADB_REBOOT_RETRIES)
     }
     await B.delay(2000); // let the emu finish stopping;
     await this.setDeviceProperty('sys.boot_completed', 0);
+    await this.root();
```

Comment:
The design of this method is messy. Please change it so the flow looks like
- check if root and assign the result to `isRoot`
- if `!isRoot` then elevate to root
- execute all further actions in `try` block
- in `finally` block check if the user was root
- execute `unroot` if the user was not root before the method has been entered

Reply:
@mykola-mokhnach the problem is the emulator losing root after stops so we need to run it again.